src/transformers/trainer.py

```diff
@@ -2419,7 +2419,13 @@ def _save_optimizer_and_scheduler(self, output_dir):
         elif self.is_deepspeed_enabled:
             # under zero3 model file itself doesn't get saved since it's bogus! Unless deepspeed
             # config `stage3_gather_16bit_weights_on_model_save` is True
-            self.model_wrapped.save_checkpoint(output_dir)
+            accept_exclude_frozen_parameters = "exclude_frozen_parameters" in set(
+                inspect.signature(self.model_wrapped.save_checkpoint).parameters.keys()
+            )
+            if accept_exclude_frozen_parameters and is_peft_available() and isinstance(self.model, PeftModel):
```

Comment:
As this is repeated everywhere, let's abstract out this check to a module level function

```py
def _is_peft_model(model):
    return is_peft_available() and isinstance(self.model, PeftModel)
```


```suggestion
            if accept_exclude_frozen_parameters and _is_peft_model(model):
```

Reply:
done